sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -861,7 +865,13 @@ class Analyzer(
           case other => i.copy(table = other)
         }
 
-      case u: UnresolvedRelation => resolveRelation(u)
+      case u: UnresolvedRelation =>
+        val relationToLogicalPlanMaps = AnalysisContext.get.relationToLogicalPlanMaps
+        relationToLogicalPlanMaps.getOrElse(u, {
```

Comment:
I think it's safer to use the qualified table name as the key, i.e. after calling `withNewNamespace`